WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -208,46 +271,55 @@ private extension PointOfSaleDashboardViewModel {
             }
         }.assign(to: &$showsCardReaderSheet)
     }
-}
 
-import enum Yosemite.OrderAction
-import struct Yosemite.Order
-@MainActor
-private extension PointOfSaleDashboardViewModel {
-    private func createOrUpdateTestOrder() async throws -> Order {
-        if let order {
-            let updatedOrder = try await update(order: order, amount: formattedOrderTotalPrice ?? "15.00")
-            self.order = updatedOrder
-            return updatedOrder
-        } else {
-            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
+    @MainActor
+    private func syncOrder(for cartProducts: [CartItem]) async {
+        guard isSyncingOrder == false else {
+            return
+        }
+        isSyncingOrder = true
+        let cart = cartProducts
+            .map {
+                POSCartItem(itemID: nil,
+                            product: $0.item,
+                            quantity: Decimal($0.quantity))
+            }
+        defer {
+            self.isSyncingOrder = false
+        }
+        do {
+            self.isSyncingOrder = true
+            let order = try await self.orderService.syncOrder(cart: cart,
+                                                              order: self.order,
+                                                              allProducts: self.items)
             self.order = order
-            return order
+            // TODO: this is temporary solution
```

Comment:
This is just here to trigger payment temporary. We will have a proper solution in next PRs

Reply:
The intention with the async task here was that if called from SwiftUI directly, it would be cancelled when the view went offscreen. It's fine if we can't do that because of the dependency on creating/updating the order, but we'll still need to make sure that cancel gets called.